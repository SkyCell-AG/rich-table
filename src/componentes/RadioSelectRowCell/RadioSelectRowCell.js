import React, {
    useCallback,
} from 'react'
import PropTypes from 'prop-types'
import {
    Radio,
} from '@mui/material'

import styles from './RadioSelectRowCell.module.css'

const propTypes = {
    checked: PropTypes.bool,
    onChange: PropTypes.func.isRequired,
}

const defaultProps = {
    checked: false,
}

const RadioSelectRowCell = ({
    checked,
    onChange,
}) => {
    const stopDetailPanelPropagation = useCallback((e) => {
        e.stopPropagation()
    }, [])

    return (
        <div
            className={styles.wrapper}
            onClick={stopDetailPanelPropagation}
            onKeyDown={stopDetailPanelPropagation}
        >
            <Radio
                checked={checked}
                onChange={onChange}
            />
        </div>
    )
}

RadioSelectRowCell.propTypes = propTypes
RadioSelectRowCell.defaultProps = defaultProps

export default RadioSelectRowCell
