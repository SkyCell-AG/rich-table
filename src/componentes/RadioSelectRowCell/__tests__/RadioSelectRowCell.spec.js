import React from 'react'
import {
    shallow,
} from 'enzyme'
import {
    Radio,
} from '@mui/material'

import RadioSelectRowCell from '../RadioSelectRowCell'

describe('RadioSelectRowCell', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(
            <RadioSelectRowCell
                checked
                onChange={jest.fn()}
            />,
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('not selected', () => {
        const wrapper = shallow(
            <RadioSelectRowCell
                onChange={jest.fn()}
            />,
        )

        expect(wrapper.find(Radio).props().checked).toEqual(false)
    })
})
