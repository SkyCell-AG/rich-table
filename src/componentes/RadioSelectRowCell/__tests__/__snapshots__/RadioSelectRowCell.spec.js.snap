// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSelectRowCell should match snapshot 1`] = `
<div
  className="wrapper"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ForwardRef(Radio)
    checked={true}
    onChange={[MockFunction]}
  />
</div>
`;
